server/src/main/java/org/opensearch/action/admin/indices/cache/clear/ClearIndicesCacheRequest.java

```diff
@@ -59,6 +61,8 @@ public ClearIndicesCacheRequest(StreamInput in) throws IOException {
         fieldDataCache = in.readBoolean();
         fields = in.readStringArray();
         requestCache = in.readBoolean();
+        requestCacheOnDisk = in.readBoolean();
```

Comment:
This should also be wrapped in a check for version being on or after 3.0.0. (Apparently nodes can have different versions in the same cluster and they could send objects like this between them. So if an old-versioned node sent this new node a ClearIndicesCacheRequest, the new one would try to read booleans that weren't written, and cause an exception)

Reply:
yep, great catch, going with 2_12 though since that is what we are trying to aim for